aws-storage-s3/src/androidTest/java/com/amplifyframework/storage/s3/AWSS3StorageDownloadAccessLevelTest.java

```diff
@@ -0,0 +1,232 @@
+
+        // This one should be unreachable, since first attempt fails
+        testDownload(downloadFile, StorageAccessLevel.PRIVATE, identityIds.get(USER_NAME_TWO));
+    }
+
+    /**
+     * Test downloading with protected access after signing in.
+     *
+     * @throws Exception if download is unsuccessful
+     */
+    @Test
+    public void testDownloadAuthenticatedProtectedAccess() throws Exception {
+        signInAs(USER_NAME_ONE);
+        testDownload(downloadFile, StorageAccessLevel.PROTECTED, getIdentityId());
+
+        signInAs(USER_NAME_TWO);
+        testDownload(downloadFile, StorageAccessLevel.PROTECTED, getIdentityId());
+    }
+
+    /**
+     * Test downloading with private access after signing in.
+     *
+     * @throws Exception if download is unsuccessful
+     */
+    @Test
+    public void testDownloadAuthenticatedPrivateAccess() throws Exception {
+        signInAs(USER_NAME_ONE);
+        testDownload(downloadFile, StorageAccessLevel.PRIVATE, getIdentityId());
+
+        signInAs(USER_NAME_TWO);
+        testDownload(downloadFile, StorageAccessLevel.PRIVATE, getIdentityId());
+    }
+
+    /**
+     * Test downloading with protected access after signing in
+     * as another user.
+     *
+     * A protected resource is READ-ONLY to all users. Download is
+     * allowed even if signed-in user does not own the resource.
+     *
+     * @throws Exception if download is unsuccessful
+     */
+    @Test
+    public void testDownloadDifferentUsersProtectedAccess() throws Exception {
+        signInAs(USER_NAME_ONE);
+        testDownload(downloadFile, StorageAccessLevel.PROTECTED, identityIds.get(USER_NAME_TWO));
+
+        signInAs(USER_NAME_TWO);
+        testDownload(downloadFile, StorageAccessLevel.PROTECTED, identityIds.get(USER_NAME_ONE));
+    }
+
+    /**
+     * Test downloading with private access after signing in
+     * as another user.
+     *
+     * Private resources are only accessible to owners. This test
+     * will throw an exception.
+     *
+     * @throws Exception if download is unsuccessful
+     */
+    @Test(expected = StorageException.class)
+    public void testDownloadDifferentUsersPrivateAccess() throws Exception {
+        signInAs(USER_NAME_ONE);
+        testDownload(downloadFile, StorageAccessLevel.PRIVATE, identityIds.get(USER_NAME_TWO));
+
+        // This part of the test should be unreachable
+        signInAs(USER_NAME_TWO);
+        testDownload(downloadFile, StorageAccessLevel.PRIVATE, identityIds.get(USER_NAME_ONE));
+    }
+
+    private void testDownload(
+            File downloadTo,
+            StorageAccessLevel accessLevel,
+            String identityId
+    ) throws Exception {
+        StorageDownloadFileOptions options = StorageDownloadFileOptions.builder()
+                .accessLevel(accessLevel)
+                .targetIdentityId(identityId)
+                .build();
+        Await.<StorageDownloadFileResult, StorageException>result(
```

Comment:
There a bunch of these similar structures throughout these new tests, just rip them all out and stuff them into a re-usable `SynchronousStorage` test utility. It should probably go into `aws-storage-s3` not `testutils` (cause I'm not sure what other type of category/plugin would ever use it.)